curve25519/src/impl_hacl.rs

```diff
@@ -0,0 +1,18 @@
+use super::*;
+
+/// Implementation of Curve25519 backed by Hacl.
+pub struct HaclCurve25519;
+
+impl Curve25519 for HaclCurve25519 {
+    fn secret_to_public(pk: &mut [u8; PK_LEN], sk: &[u8; SK_LEN]) {
+        crate::hacl::secret_to_public(pk, sk)
+    }
+
+    fn ecdh(out: &mut [u8; SHK_LEN], pk: &[u8; PK_LEN], sk: &[u8; SK_LEN]) -> Result<(), Error> {
```

Comment:
The glue code that's calling into the hacl code should have enough comments to understand how it ensures that all the preconditions of the underlying, verified code are ensured. Since we're in all Rust, that's relatively easy. But let's make sure that we have comments on the glue explaining how it works.

Reply:
Can you expand on what you mean? I am just passing on arguments. I could say something like "the documentation of HACL says that this should be a 64-byte buffer and I pass in a [u8; 64], so the type system enforces that the inputs are correct", but I am not sure that is what you mean.

Reply:
Yes, the hacl code should have pretty good comments already on that. I just want to make sure we have a little in here so that we know what's going on. Thanks for adding some.
When you'd use doc comments, they'd show up in the IDE.